pkg/controller/controller.go

```diff
@@ -144,6 +153,12 @@ func (s *Service) CreateVolume(ctx context.Context, req *csi.CreateVolumeRequest
 				nasName: arr.GetNasName(),
 			}
 		}
+
+		if params[common.KeyNfsACL] != "" {
+			nfsAcls = params[common.KeyNfsACL] // Storage class takes precedence
+		} else if arr.NfsAcls != "" {
```

Comment:
if params[common.KeyNfsACL] and arr.NfsAcls are both set, the logic gives precedence to secret. Please check

Reply:
If both are set, params[common.KeyNfsACL] is checked first and that is set. This params[common.KeyNfsACL] is from storage class. So it looks good to me.